spec/ruby/core/module/private_spec.rb

```diff
@@ -51,4 +51,43 @@ def foo; end
       Module.new.send(:private, :undefined)
     end.should raise_error(NameError)
   end
+
+  it "only makes the method private in the class it is called on" do
+    base = Class.new do
+      def wrapped
+        1
+      end
+    end
+
+    klass = Class.new(base) do
+      def wrapped
+        super + 1
+      end
+      private :wrapped
+    end
+
+    base.new.wrapped.should == 1
+    lambda do
+      klass.new.wrapped
+    end.should raise_error(NameError)
+  end
+
+  it "continues to allow a prepended module method to call super" do
+    wrapper = Module.new do
+      def wrapped
+        super + 1
+      end
+    end
+
+    klass = Class.new do
+      prepend wrapper
+
+      def wrapped
+        1
+      end
+      private :wrapped
+    end
+
+    klass.new.wrapped.should == 2
```

Comment:
Previously this failed with

```
NoMethodError: super: no superclass method `wrapped' for #<#<Class:0x00007fa541046210>:0x00007fa5410460f8>
```

because it was adding a [method that called `super`](https://github.com/ruby/ruby/blob/2697ae451d67547ed601bce8e05b9779123142de/vm_method.c#L1080) which is meant for the case where an ancestor defines the method rather than when it is defined in a prepended module.